### PR TITLE
Allow gamemode string in server.properties

### PR DESCRIPTION
**Pull request by JackNoordhuis**
Addresses #2692.

### CODE REVIEW

**Comment by dktapps** [comment]
There is no error checking in this PR, which will cause the server to crash if an invalid gamemode is found in server.properties. This is why I did not apply this change myself, as trivial as it looks.

**Comment by JackNoordhuis** [comment]
I wasn't sure what behaviour to implement for catching the error. I don't think the server should start given a bad configuration but a warning and defaulting back to survival could be an alternative?

**Comment by dktapps** [comment]
The configuration crashing issue is a more general problem. The point is that I explicitly avoided applying this change myself because of the additional problems it would cause without a proper solution for config handling.

File: src/pocketmine/Server.php [hunk 1]
```diff
@@ -466,7 +466,7 @@ public function isHardcore() : bool{
 	 * @return int
 	 */
 	public function getDefaultGamemode() : int{
-		return $this->getConfigInt("gamemode", 0) & 0b11;
+		return GameMode::fromString($this->getConfigString("gamemode", "s"));
```
Comment (dktapps) [inline src/pocketmine/Server.php]: I would be happier if `(string) GameMode::SURVIVAL` was used here. It's too easy to miss this if refactors happen.

**Review by dktapps — CHANGES_REQUESTED** [review]

File: src/pocketmine/Server.php [hunk 2]
```diff
@@ -430,7 +430,7 @@ public function getServerUniqueId(){
 	 * @return int
 	 */
 	public function getGamemode() : int{
-		return $this->getConfigInt("gamemode", 0) & 0b11;
+		return GameMode::fromString($this->getConfigString("gamemode", "s"));
```
Comment (dktapps) [inline src/pocketmine/Server.php]: and here too (ofc)

**Comment by dktapps** [comment]
Closing due to lack of activity on review comments.

**Pull request closed without merging**